packages/client/src/workflow-client.ts

```diff
@@ -640,6 +712,114 @@ export class WorkflowClient extends BaseClient {
     return await decodeFromPayloadsAtIndex(this.dataConverter, 0, response.queryResult?.payloads);
   }
 
+  protected _makeStartUpdateHandler(waitForStage: temporal.api.enums.v1.UpdateWorkflowExecutionLifecycleStage) {
```

Comment:
Is the only reason for this function to pass `waitForStage`?

Reply:
Yes, that's right. I don't think the stage should be available to the user in the interceptor input. For example, if a user is doing `executeUpdate` it does not make sense for the stage to be anything other than `COMPLETED`. I don't like the closure but I couldn't think of a cleaner solution without extensive changes.

Reply:
> I don't think the stage should be available to the user in the interceptor input. For example, if a user is doing executeUpdate it does not make sense for the stage to be anything other than COMPLETED

Make sense. In that case, I would keep this function "simple" (without the outer wrapper, with waitForStage as the first arg), and make the wrapper at call site instead. Can even merge that with the existing `.bind()` call.

For example, instead of:

```
const next = this._makeStartUpdateHandler(waitForStage).bind(this);
```

you could do

```
const next = this._startUpdateHandler().bind(this, waitForStage);
```

Reply:
Perfect, thanks! It didn't occur to me to think of using `.bind(...)` for something other than `.bind(this)`.